bespin_api_v2/tests_api.py

```diff
@@ -5,145 +5,181 @@
-        url = reverse('v2-admin_workflow-list')
+        url = self.list_url()
         response = self.client.get(url, format='json')
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
 
     def test_list_with_admin_user(self):
-        workflow = Workflow.objects.create(name='Exome Seq', tag='exomeseq')
+        model_object = self.create_model_object()
         self.user_login.become_admin_user()
-        url = reverse('v2-admin_workflow-list')
+        url = self.list_url()
         response = self.client.get(url, format='json')
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 1)
-        self.assertEqual(response.data[0]['id'], workflow.id)
-        self.assertEqual(response.data[0]['name'], 'Exome Seq')
-        self.assertEqual(response.data[0]['tag'], 'exomeseq')
+        self.check_list_response(model_object, response.data)
 
     def test_retrieve_with_admin_user(self):
-        workflow = Workflow.objects.create(name='Exome Seq', tag='exomeseq')
+        model_object = self.create_model_object()
         self.user_login.become_admin_user()
-        url = reverse('v2-admin_workflow-list') + str(workflow.id) + '/'
-        response = self.client.get(url, format='json')
+        url = self.object_url(model_object.id)
+        response = self.client.get(url ,format='json')
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(response.data['id'], workflow.id)
-        self.assertEqual(response.data['name'], 'Exome Seq')
-        self.assertEqual(response.data['tag'], 'exomeseq')
+        self.check_single_response(model_object, response.data)
 
     def test_create_with_admin_user(self):
         self.user_login.become_admin_user()
-        url = reverse('v2-admin_workflow-list')
-        response = self.client.post(url, format='json', data={
-            'name': 'Exome Seq',
-            'tag': 'exomeseq',
-        })
+        url = self.list_url()
+        response = self.client.post(url, format='json', data=self.build_post_data())
         self.assertEqual(response.status_code, status.HTTP_201_CREATED)
-        self.assertEqual(response.data['name'], 'Exome Seq')
-        self.assertEqual(response.data['tag'], 'exomeseq')
-        workflows = Workflow.objects.all()
-        self.assertEqual(len(workflows), 1)
-        self.assertEqual(workflows[0].name, 'Exome Seq')
-        self.assertEqual(workflows[0].tag, 'exomeseq')
+        model_object = self.get_model_object(response.data['id'])
+        self.check_single_response(model_object, response.data)
 
     def test_put_fails_with_admin_user(self):
-        workflow = Workflow.objects.create(name='Exome Seq', tag='exomeseq')
         self.user_login.become_admin_user()
-        url = reverse('v2-admin_workflow-list') + str(workflow.id) + '/'
-        response = self.client.post(url, format='json', data={})
+        url = self.object_url(1)
```

Comment:
The 1 key value here had me wondering if there was some code that created an item with that key. Reading the rest of this method makes sense that request is disallowed before that get's used.